src/app/shared/form/builder/ds-dynamic-form-ui/ds-dynamic-type-bind-relation.service.ts

```diff
@@ -0,0 +1,230 @@
+import { Inject, Injectable, Injector, Optional } from '@angular/core';
+import { FormControl } from '@angular/forms';
+
+import { Subscription } from 'rxjs';
+import { startWith } from 'rxjs/operators';
+
+import {
+  AND_OPERATOR,
+  DYNAMIC_MATCHERS,
+  DynamicFormControlCondition,
+  DynamicFormControlMatcher,
+  DynamicFormControlModel,
+  DynamicFormControlRelation,
+  DynamicFormRelationService, MATCH_VISIBLE,
+  OR_OPERATOR
+} from '@ng-dynamic-forms/core';
+
+import {hasNoValue, hasValue, isEmpty} from '../../../empty.util';
```

Comment:
According to LGTM, the `isEmpty` here can be removed.  It's not used in this file.

Reply:
thanks, done